src/ui_fsmenu/addons/manager.cc

```diff
@@ -1178,13 +1178,14 @@ void AddOnsCtrl::update_dependency_errors() {
 						too_late = true;
 					}
 				}
-				assert(prev != nullptr);
-				assert(!too_late || next != nullptr);
-				if (too_late && AddOns::order_matters(prev->category, next->category)) {
-					warn_requirements.push_back(format(
-					   _("· ‘%1$s’ requires first ‘%2$s’ and then ‘%3$s’, but they are "
-					     "listed in the wrong order"),
-					   addon->first->descname(), prev->descname(), search_result->first->descname()));
+				if (prev != nullptr) {
+					assert(!too_late || next != nullptr);
+					if (too_late && AddOns::order_matters(prev->category, next->category)) {
+						warn_requirements.push_back(format(
+						   _("· ‘%1$s’ requires first ‘%2$s’ and then ‘%3$s’, but they are "
+						     "listed in the wrong order"),
+						   addon->first->descname(), prev->descname(), search_result->first->descname()));
+					}
```

Comment:
Shouldn't this also have an else case for the former assert failure? Or wasn't it an assert because it's also a problem?

Reply:
If `prev` is `nullptr` this means that the add-on `prev` is not installed. In this case we don't need to check the order between `prev` and the current add-on – it can't be misordered since `prev` is not installed at all, and the warning that `prev` is missing will already have been generated in the previous iteration in the generic check for add-on installs above.

Reply:
I couldn't fully follow it through, but I take your word for it. :)